ui/app/AppLayouts/Profile/views/wallet/EditNetworkForm.qml

```diff
@@ -0,0 +1,193 @@
+        property var evaluateRpcEndPoint: Backpressure.debounce(root, 400, function (value) {
+            root.evaluateEndPoint(value)
+        })
+
+        function revertValues() {
+            warningCheckbox.checked = false
+            d.evaluationStatus = EditNetworkForm.UnTouched
+            d.evaluationStatusFallBackRpc = EditNetworkForm.UnTouched
+            mainRpcInput.text = network.rpcURL
+            failoverRpcUrlInput.text = network.fallbackURL
+        }
+    }
+
+    onVisibleChanged: if(!visible) {d.revertValues()}
+
+    Connections {
+        target: networksModule
+        function onUrlVerified(url, status) {
+            if(url === mainRpcInput.text) {
+                d.evaluationStatus = status
+            }
+            else if(url === failoverRpcUrlInput.text) {
+                d.evaluationStatusFallBackRpc = status
+            }
+        }
+    }
+
+    spacing: 20
+
+    StatusInput {
+        Layout.fillWidth: true
+        label: qsTr("Network name")
+        text: !!network ? network.chainName : ""
+        input.enabled: false
+    }
+
+    StatusInput {
+        Layout.fillWidth: true
+        label: qsTr("Short name")
+        text: !!network ? network.shortName : ""
+        input.enabled: false
+    }
+
+    StatusInput {
+        Layout.fillWidth: true
+        label: qsTr("Chain ID")
+        text: !!network ? network.chainId : ""
+        input.enabled: false
+    }
+
+    StatusInput {
+        Layout.fillWidth: true
+        label: qsTr("Native Token Symbol")
+        text: !!network ? network.nativeCurrencySymbol : ""
+        input.enabled: false
+    }
+
+    Item {
+        Layout.fillWidth: true
+        Layout.preferredHeight: childrenRect.height
+        StatusBaseText {
+            id: requiredText
+            anchors.top: parent.top
+            anchors.topMargin: 4
+            anchors.right: parent.right
+            elide: Text.ElideRight
+            text: qsTr("Required")
+            font.pixelSize: 12
+            color: Theme.palette.baseColor1
+        }
+        StatusInput {
+            id: mainRpcInput
+            width: parent.width
+            label: qsTr("Main JSON RPC URL")
+            text: !!network ? network.rpcURL : ""
+            onTextChanged: {
+                if(!!text && text !== network.rpcURL) {
+                    d.evaluationStatus = EditNetworkForm.Pending
+                    Qt.callLater(d.evaluateRpcEndPoint, text);
+                }
+            }
+            errorMessageCmp.horizontalAlignment: d.evaluationStatus === EditNetworkForm.Pending ||
+                                                 d.evaluationStatus === EditNetworkForm.Verified ?
+                                                     Text.AlignLeft: Text.AlignRight
+            errorMessageCmp.visible: d.evaluationStatus !== EditNetworkForm.UnTouched
+            errorMessageCmp.text: d.evaluationStatus === EditNetworkForm.Pending ?
```

Comment:
Please turn this into a series of `if` statements :) This is unreadable

Even better, make `evaluation` status an enum

Reply:
it is already an enum, I put it in a function with switch case now :)